include/iser-private.h

```diff
@@ -195,6 +195,7 @@ struct iser_conn {
 	int                          qp_max_recv_dtos;
 	int                          min_posted_rx;
 	uint16_t                     max_cmds;
+	uint8_t                      cmthread_should_run;
```

Comment:
Please use a mutex (pthread_mutex_t) + condition variable (pthread_cond_t) in addition to this flag.

Reply:
On second thought, I'm not sure introducing a pthread mutex and condition variable is essential.

Reply:
Because libiscsi uses RDMA RC, client/server need send/receive DisconnectRequest & DisconnectReply. During disconnecting, the network may hit packet dropping case. 
I'm not sure the cm thread can receive the disconnect event or not, it's hard to emulate this case.
On the other hand, qemu locks mutex, then reconnect, finally unlocks mutex. If we blocks in reconnect for a long time, qemu will hit lock racing. So I add a timeout here.

In fact, I tested for 100 times, 1 case actually hit a timeout.